src/com/minseung/datastructure/StackAndQueue.java

```diff
@@ -0,0 +1,75 @@
+package com.minseung.datastructure;
+
+import java.util.EmptyStackException;
+
+class Stack<T> {
+
+    //TODO: Could I make Node class in a different file, and use the class in Stack class & LinkedList class too?
+    static class Node<T> {
+        private T data;
+        private Node<T> next = null;
+
+        Node(T data) {
+            this.data = data;
+        }
+
+        void setNext(Node<T> node) {
+            this.next = node;
+        }
+    }
+
+    private Node<T> top = null;
+
+    public void push(T data) {
+        Node<T> node = new Node<>(data);
+
+        node.next = top;
+        top = node;
+    }
+
+    public Node<T> pop() {
+        if (top == null) {
+            throw new EmptyStackException();
```

Comment:
Would you add test cases for the exceptions? L42 as well.

You can find an example from the first answer of the link below:
https://stackoverflow.com/questions/156503/how-do-you-assert-that-a-certain-exception-is-thrown-in-junit-4-tests